src/auspex/instruments/holzworth.py

```diff
@@ -73,23 +73,35 @@ def frequency(self):
         return float(v.split()[0])*1e-3
     @frequency.setter
     def frequency(self, value):
-        self.ch_query(":FREQ:{} GHz".format(value))
+        fmin = float((self.ch_query(":FREQ:MIN?")).split()[0])
+        fmax = float((self.ch_query(":FREQ:MAX?")).split()[0])
+        if fmin*1e-3 <= value <= fmax*1e-3:
+            self.ch_query(":FREQ:{} GHz".format(value))
+        else:
+            err_msg = "The value {} GHz is outside of the allowable range {}-{} GHz specified for instrument '{}'.".format(value, fmin*1e-3, fmax*1e-3, self.name)
```

Comment:
I'm not we we want to require Python 3.6 but the [f-strings](https://docs.python.org/3/reference/lexical_analysis.html#f-strings) are much nicer for this.